src/target/esp32c3/src/clock.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <target/clock.h>
+#include <soc_utils.h>
+#include <soc/system_reg.h>
+#include <soc/soc.h>
+#include <soc/uart_reg.h>
+
+#define MHZ 1000000
+
+extern uint32_t esp_rom_get_xtal_freq(void);
+extern void esp_rom_set_cpu_ticks_per_us(uint32_t ticks_per_us);
+extern uint32_t esp_rom_get_apb_freq(void);
+
+static uint32_t s_cpu_freq = 0;
+
+void stub_target_clock_init(void)
+{
+    REG_SET_FIELD(SYSTEM_SYSCLK_CONF_REG, SYSTEM_SOC_CLK_SEL, 1);
+    REG_SET_FIELD(SYSTEM_CPU_PER_CONF_REG, SYSTEM_PLL_FREQ_SEL, 1);
+    REG_SET_FIELD(SYSTEM_CPU_PER_CONF_REG, SYSTEM_CPUPERIOD_SEL, 1);
+    s_cpu_freq = 160 * MHZ;
+    esp_rom_set_cpu_ticks_per_us(s_cpu_freq / MHZ);
+}
+
+uint32_t stub_target_get_cpu_freq(void)
+{
+    if (s_cpu_freq == 0) {
```

Comment:
Do you want to check if `stub_target_clock_init` is called? If not called, will it be different from the value that set in clock init?